server/src/test/java/org/elasticsearch/health/node/HealthInfoTests.java

```diff
@@ -23,60 +22,49 @@ protected Writeable.Reader<HealthInfo> instanceReader() {
+        var repositoriesInfoByNode = randomMap(0, 10, () -> tuple(randomAlphaOfLength(10), randomRepoHealthInfo()));
+        return new HealthInfo(diskInfoByNode, randomBoolean() ? randomDslHealthInfo() : null, repositoriesInfoByNode);
     }
 
     @Override
     public HealthInfo mutateInstance(HealthInfo originalHealthInfo) {
-        Map<String, DiskHealthInfo> diskHealthInfoMap = originalHealthInfo.diskInfoByNode();
-        Map<String, DiskHealthInfo> diskHealthInfoMapCopy = new HashMap<>(diskHealthInfoMap);
-        if (diskHealthInfoMap.isEmpty()) {
-            diskHealthInfoMapCopy.put(
-                randomAlphaOfLength(10),
-                new DiskHealthInfo(randomFrom(HealthStatus.values()), randomFrom(DiskHealthInfo.Cause.values()))
-            );
-        } else {
-            switch (randomIntBetween(1, 3)) {
-                case 1 -> {
-                    diskHealthInfoMapCopy.put(
-                        randomAlphaOfLength(10),
-                        new DiskHealthInfo(randomFrom(HealthStatus.values()), randomFrom(DiskHealthInfo.Cause.values()))
-                    );
-                }
-                case 2 -> {
-                    String someNode = randomFrom(diskHealthInfoMap.keySet());
-                    diskHealthInfoMapCopy.put(
-                        someNode,
-                        new DiskHealthInfo(
-                            randomValueOtherThan(diskHealthInfoMap.get(someNode).healthStatus(), () -> randomFrom(HealthStatus.values())),
-                            randomFrom(DiskHealthInfo.Cause.values())
-                        )
-                    );
-                }
-                case 3 -> {
-                    diskHealthInfoMapCopy.remove(randomFrom(diskHealthInfoMapCopy.keySet()));
-                }
-                default -> throw new IllegalStateException();
-            }
-        }
+        return mutateHealthInfo(originalHealthInfo);
+    }
+
+    public static HealthInfo mutateHealthInfo(HealthInfo originalHealthInfo) {
+        var randomInt = randomInt(2);
         return new HealthInfo(
-            diskHealthInfoMapCopy,
-            randomValueOtherThan(originalHealthInfo.dslHealthInfo(), HealthInfoTests::randomDslHealthInfo)
+            randomInt == 0
+                ? mutateMap(originalHealthInfo.diskInfoByNode(), () -> randomAlphaOfLength(10), HealthInfoTests::randomDiskHealthInfo)
+                : originalHealthInfo.diskInfoByNode(),
+            randomInt == 1
+                ? randomValueOtherThan(originalHealthInfo.dslHealthInfo(), HealthInfoTests::randomDslHealthInfo)
+                : originalHealthInfo.dslHealthInfo(),
+            randomInt == 2
+                ? mutateMap(
+                    originalHealthInfo.repositoriesInfoByNode(),
+                    () -> randomAlphaOfLength(10),
+                    HealthInfoTests::randomRepoHealthInfo
+                )
+                : originalHealthInfo.repositoriesInfoByNode()
```

Comment:
May I suggest a different control flow? I personally like this way of writing mutators:

https://github.com/elastic/elasticsearch/blob/71c3f34ce5c50cf04c35c1a65a4c15e78672949e/modules/data-streams/src/test/java/org/elasticsearch/datastreams/action/GetDataStreamsResponseTests.java#L294-L324

Let me know what you think.